### DISCUSSION TITLE
How to change Jupyter runtime directory in extension settings?

### DISCUSSION THREAD

**Discussion by fmungai1** [microsoft/vscode-jupyter#13304] [Questions and Answers]
I am having trouble activating Jupyter runtime environment on a remote Linux server. I am getting the following error (no space on device):

> Traceback (most recent call last): File "/user-workspace/.vscode-server/extensions/ms-toolsai.jupyter-2022.1.1301854968/pythonFiles/vscode_datascience_helpers/daemon/daemon_python.py", line 54, in _decorator return func(self, *args, **kwargs) File "/user-workspace/.vscode-server/extensions/ms-toolsai.jupyter-2022.1.1301854968/pythonFiles/vscode_datascience_helpers/jupyter_daemon.py", line 108, in m_exec_module_observable self._start_notebook(args, cwd, env) File "/user-workspace/.vscode-server/extensions/ms-toolsai.jupyter-2022.1.1301854968/pythonFiles/vscode_datascience_helpers/jupyter_daemon.py", line 162, in _start_notebook app.launch_new_instance() File "/user-workspace/py3.6/lib64/python3.6/site-packages/jupyter_core/application.py", line 264, in launch_instance return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs) File "/user-workspace/py3.6/lib64/python3.6/site-packages/traitlets/config/application.py", line 664, in launch_instance app.start() File "/user-workspace/py3.6/lib64/python3.6/site-packages/notebook/notebookapp.py", line 2338, in start self.write_browser_open_file() File "/user-workspace/py3.6/lib64/python3.6/site-packages/notebook/notebookapp.py", line 2241, in write_browser_open_file with open(self.browser_open_file, 'w', encoding='utf-8') as f: OSError: [Errno 28] No space left on device: '/home/user/.local/share/jupyter/runtime/nbserver-31539-open.html'

I have tried setting the environment variables `JUPYTER_CONFIG_DIR`, `JUPYTER_DATA_DIR` and `JUPYTER_RUNTIME_DIR` in bashrc but still not working.

So I am trying to change the directory where the runtime will save the necessary files (There is a separate drive with more space). I can't find any setting in the extension to change this. Please assist. @DonJayamanne

**Comment by kushaangupta** [comment]
Any update on this? I am able to set **`JUPYTER_RUNTIME_DIR`** using `.env` file, but not sure why most of the other ways don't work.  @fmungai1 @DonJayamanne
